src/scripts/import-needs-assessment-data/types.d.ts

```diff
@@ -0,0 +1,56 @@
+export enum UploadWorkflowStatus {
+  PROCESSING = "PROCESSING",
+  ORIGINAL_DATA_INVALID = "ORIGINAL_DATA_INVALID",
+  DUPLICATE_CHECK_ERROR = "DUPLICATE_CHECK_ERROR",
+  ALREADY_EXISTS = "ALREADY_EXISTS",
+  UPLOAD_ERROR = "UPLOAD_ERROR",
+  SUCCESS = "SUCCESS",
+  OTHER = "OTHER",
+}
+
+export type UploadWorkflow<T> = {
+  data: T;
+  orig: string | T; // Union type allowing either a csv string or parsed JSON object
+  status: UploadWorkflowStatus;
+  logs: string[];
+};
+
+export type UploadWorkflowResults<T> = {
+  [Property in keyof UploadWorkflowStatus]: T[];
+};
+
+export type RegionUploadWorkflow = UploadWorkflow<Region>;
+export type RegionUploadWorkflowResults = UploadWorkflowResults<Region>;
+export type SubregionUploadWorkflow = UploadWorkflow<Subregion>;
+export type SubregionUploadWorkflowResults = UploadWorkflowResults<Subregion>;
+
+export type Region = {
+  region?: string;
+};
+
+export type Subregion = {
+  subregion?: string;
+};
+
+export type NeedAssessment = {
+  id?: string;
+  survey?: {
+    id?: string;
+    year?: string;
+    quarter?: string;
+    url?: string;
```

Comment:
Maybe for later, but if we're declaring separate taxonomies that mirror what's in Strapi, we could break it out also into scripts, sort of like

```js
export type Item = {
// item description from product import
}

export type Survey = {
   id: string;
... etc
}
export type NeedsAssessment = {
id: String,
survey?:  Survey,
...}
```
and so on

Then each part of the pipeline could export from there and changes could stay synced.

Reply:
@veryeli def agree!  Sounds like a good refactor.